### PR TITLE
[ICDS Dashboard]: Fix the Agg Child Health Helper

### PR DESCRIPTION
**Pull request by ShubhamBansal1997**
Added the valid_in_month parameter for the stunting indicator in the agg child health helper

##### SUMMARY

https://dimagi-dev.atlassian.net/browse/ICDS-1240

### CODE REVIEW

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
hey, in spite of asking up an update here. I think it makes more sense to make this change in the underlying table itself which is the child_health monthly table. That will automatically have affect on the dependent tables and it will be more central change

File: custom/icds_reports/utils/aggregation_helpers/distributed/child_health_monthly.py [hunk 1]
```diff
@@ -420,8 +420,11 @@ def create_partition(self, state_id):
 
     def aggregation_queries(self):
         return [
-            """INSERT INTO "{new_tablename}" (SELECT * FROM "{tmp_tablename}")""".format(new_tablename=self.new_tablename, tmp_tablename=self.temporary_tablename),
+            """INSERT INTO "{new_tablename}" (SELECT * FROM "{tmp_tablename}")""".format(
+                new_tablename=self.new_tablename, tmp_tablename=self.temporary_tablename),
             'DROP TABLE IF EXISTS "{monthly_tablename}"'.format(monthly_tablename=self.monthly_tablename),
-            """ALTER TABLE "{new_tablename}" RENAME TO \"{tablename}\"""".format(new_tablename=self.new_tablename, tablename=self.monthly_tablename),
-            """ALTER TABLE "{tablename}" ATTACH PARTITION "{monthly_tablename}" FOR VALUES IN ('{month}')""".format(monthly_tablename=self.monthly_tablename, month=self.month.strftime('%Y-%m-%d'), tablename=self.tablename),
+            """ALTER TABLE "{new_tablename}" RENAME TO \"{tablename}\"""".format(
```
Comment (Rohit25negi) [inline custom/icds_reports/utils/aggregation_helpers/distributed/child_health_monthly.py]: this change looks not required. can you ensure to revert these ones
@ShubhamBansal1997

File: custom/icds_reports/utils/aggregation_helpers/distributed/child_health_monthly.py [hunk 2]
```diff
@@ -261,8 +261,8 @@ def _state_aggregation_query(self, state_id):
                 "ELSE 'unmeasured' END".format(height_eligible)),
             ("zscore_grading_hfa", "gm.zscore_grading_hfa"),
             ("zscore_grading_hfa_recorded_in_month",
-                "CASE WHEN (date_trunc('MONTH', gm.zscore_grading_hfa_last_recorded) = %(start_date)s) "
-                "THEN 1 ELSE 0 END"),
+                "CASE WHEN (date_trunc('MONTH', gm.zscore_grading_hfa_last_recorded) = %(start_date)s) AND {} "
```
Comment (Rohit25negi) [inline custom/icds_reports/utils/aggregation_helpers/distributed/child_health_monthly.py]: can you also add this constraint to `zscore_grading_hfa` column.

**Review by Rohit25negi — CHANGES_REQUESTED** [review]

File: custom/icds_reports/utils/aggregation_helpers/distributed/child_health_monthly.py (hunk 2 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/utils/aggregation_helpers/distributed/child_health_monthly.py]: Updated

**Review by Rohit25negi — APPROVED** [review]

**Comment by Rohit25negi** [comment]
@ShubhamBansal1997  are we also planning to correct historical data for this?

**Comment by calellowitz** [comment]
@ShubhamBansal1997 @Rohit25negi , I am reverting this PR. I am not sure I understand how this resolves the issue in the ticket or that this is correct behavior, and I think it is possible that this will have some unintended consequences. We might want to have a discussion about this before figuring out what the correct fix is.

**Comment by Rohit25negi** [comment]
> @ShubhamBansal1997 @Rohit25negi , I am reverting this PR. I am not sure I understand how this resolves the issue in the ticket or that this is correct behavior, and I think it is possible that this will have some unintended consequences. We might want to have a discussion about this before figuring out what the correct fix is.

Sure, Though we have discussed this with Nitish before implementing this change. I can ask him to revert to my Jira comment I m going to post for the same.

**Pull request merged**